src/_igraph/graphobject.c

```diff
@@ -11745,6 +11745,135 @@ PyObject *igraphmodule_Graph_community_walktrap(igraphmodule_GraphObject * self,
   return res;
 }
 
+/**
+ * Leiden community detection method of Traag, Waltman & van Eck
+ */
+PyObject *igraphmodule_Graph_community_leiden(igraphmodule_GraphObject *self,
+        PyObject *args, PyObject *kwds) {
+
+  static char *kwlist[] = {"edge_weights", "node_weights", "resolution_parameter",
+                           "normalize_resolution", "beta", "initial_membership", "n_iterations", NULL};
+  
+  PyObject *edge_weights_o = Py_None;
+  PyObject *node_weights_o = Py_None;
+  PyObject *initial_membership_o = Py_None;
+  PyObject *res;
+
+  int error = 0, i;
+  long int n_iterations = 2;
+  double resolution_parameter = 1.0;
+  double beta = 0.01;
+  igraph_vector_t *edge_weights = NULL, *node_weights = NULL, *membership;
```

Comment:
I think we could simply allocate an `igraph_vector_t membership` vector statically, no need for a dynamic allocation here. This would simplify the body of the function a bit.

Reply:
Scratch that, I was wrong, sorry. The initial membership vector comes from the arguments as a Python object so we need this to be a pointer.